cutadapt.wdl

```diff
@@ -59,26 +63,36 @@ task Cutadapt {
 
         Int cores = 1
         Int memory = 8
-        String dockerTag = "1.16--py36_2"
+        String dockerTag = "2.3--py36h14c3975_0"
     }
 
     String read2outputArg = if (defined(read2output))
         then "mkdir -p $(dirname " + read2output + ")"
         else ""
 
+    # Some WDL magic here to set both adapters with one setting.
+    # If then else's are needed to keep the variable optional and undefined
+    Array[String]+? adapterForward = if (defined(adapter) || defined(adapterBoth)) then select_first([adapter, adapterBoth]) else adapter
+    # Assume adapterRead2 will not be set when read2 is not set.
+    Array[String]+? adapterReverse = if defined(read2) then select_first([adapterRead2, adapterBoth]) else adapterRead2
```

Comment:
Even if `read2` is defined, both `adapterRead2` and `adapterBoth` may be undefined, so won't `select_first` throw an error here in such cases?

Reply:
Good catch! :+1: